src/metabase/util/honeysql_extensions.clj

```diff
@@ -208,11 +216,17 @@
 
 (defn is-of-type?
   "Is `honeysql-form` a typed form with `database-type`?
+  Where `database-type` could be a string or a regex.
+
+    (is-of-type? expr \"datetime\") ; -> true
+    (is-of-type? expr #\"int*\") ; -> true"
 
-    (is-of-type? expr \"datetime\") ; -> true"
   [honeysql-form database-type]
-  (= (some-> honeysql-form type-info type-info->db-type str/lower-case)
-     (some-> database-type name str/lower-case)))
+  (let [form-type (some-> honeysql-form type-info type-info->db-type str/lower-case)]
+    (if (instance? java.util.regex.Pattern database-type)
+      (some? (re-find database-type form-type))
+      (= form-type
+         (some-> database-type name str/lower-case)))))
```

Comment:
I have quite the same idea that I talked to Dan when I first working with convertTimezone.

I think it's much better to annotate the typeinfo using the base-type instead of the database. One example is that if we want to cover all type for convert-timezone for postgres, we have to have a regex that matches `timestamptz, timestamp with time zone, timestamp (p) with time zone`.

It's just lucky for us that the JDBC prefers to consistently call all of them `timestamptz`.